File: lab2/Fish.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab2
{
   public abstract class Fish : IAnimals
    {
        protected float startPosX;
        protected float startPosY;

        public abstract void drawAnimal(Graphics g);
        public abstract void moveAnimal(Graphics g);
       
        protected int age;
        public virtual int Age { protected set; get; }

        public void grow(int a)
        {
            if (age + a < Age)
            {
                Age++;
            }
        }


        public void setPos(int x, int y)
        {
            startPosX = x;
            startPosY = y;
        }
        public virtual double Speed { protected set; get; }
        public virtual int Weight { protected set; get; }
        public virtual Color ColorBody{ protected set; get; }
        //5
        public virtual void setMainColor(Color color)
        {
            ColorBody = color;
        }
        //
    }
}

File: src/Fish.java
import java.awt.Color;
import java.awt.Graphics;
import java.io.Serializable;

public abstract class Fish implements IAnimal,Serializable{
	protected int startPosX;
	protected int startPosY;

	public abstract void drawAnimal(Graphics g);

	public abstract void moveAnimal(Graphics g);

	public void setPos(int x, int y) {
		startPosX = x;
		startPosY = y;
	}

	public void setMainColor(Color color) {
		ColorBody = color;
	}

	protected int age;
	private int Age;

	protected void SetAge(int a) {
		Age = a;
	}

	public int GetAge() {
		return Age;
	}

	public void grow(int a) {
		if (age + a < Age) {
			age++;
		}
	}

	private double Speed;

	protected void SetSpeed(int s) {
		Speed = s;
	}

	public double GetSpeed() {
		return Speed;
	}

	private int Weight;

	protected void SetWeight(int w) {
		Weight = w;
	}

	public int GetWeight() {
		return Weight;
	}

	private Color ColorBody;

	protected void SetColorBody(Color c) {
		ColorBody = c;
	}

	public Color GetColorBody() {
		return ColorBody;
	}

}
